sdk/search/search-documents/samples/javascript/README.md

```diff
@@ -5,6 +5,51 @@ These sample programs show how to use the JavaScript client libraries for Azure
+| [createOrUpdateDataSourceConnection.js][createOrUpdateDataSourceConnection] | Updates a Datasource Connection |
+| [deleteDataSourceConnectionByName.js][deleteDataSourceConnectionByName] | Deletes Datasource Connection by Name |
+| [deleteDataSourceConnectionByObject.js][deleteDataSourceConnectionByObject] | Deletes Datasource Connection by Object |
+| [getDataSourceConnection.js][getDataSourceConnection] | Gets a Datasource Connection |
+| [listDataSourceConnectionNames.js][listDataSourceConnectionNames] | List names of Datasource Connections |
+| [listDataSourceConnections.js][listDataSourceConnections] | List Datasource Connections |
+| **Indexers** |
+| [createIndexer.js][createIndexer] | Creates an Indexer |
+| [createOrUpdateIndexer.js][createOrUpdateIndexer] | Updates an Indexer |
+| [deleteIndexerByName.js][deleteIndexerByName] | Deletes Indexer by Name |
+| [deleteIndexerByObject.js][deleteIndexerByObject] | Deletes Indexer by Object |
+| [getIndexer.js][getIndexer] | Gets an Indexer |
+| [getIndexerStatus.js][getIndexerStatus] | Gets the status of an Indexer |
+| [listIndexerNames.js][listIndexerNames] | List names of Indexers |
+| [listIndexers.js][listIndexers] | List Indexers |
+| [resetIndexer.js][resetIndexer] | Resets an Indexer |
+| [runIndexer.js][runIndexer] | Runs an Indexer |
+| **Indexes** |
+| [analyzeText.js][analyzeText] | Analyzes a given text |
+| [createIndex.js][createIndex] | Creates an Index |
+| [createOrUpdateIndex.js][createOrUpdateIndex] | Updates an Index |
+| [deleteIndexByName.js][deleteIndexByName] | Deletes Index by Name |
+| [deleteIndexByObject.js][deleteIndexByObject] | Deletes Index by Object |
+| [getIndex.js][getIndex] | Gets an Index |
+| [getIndexStatistics.js][getIndexStatistics] | Gets the Statistics of an Index |
+| [listIndexes.js][listIndexes] | List Indexes |
+| [listIndexNames.js][listIndexNames] | List Names of Indexes |
+| **Skillsets** |
+| [createOrUpdateSkillset.js][createOrUpdateSkillset] | Updates a skillset |
+| [createSkillset.js][createSkillset] | Creates a skillset |
+| [deleteSkillsetByName.js][deleteSkillsetByName] | Deletes a Skillset by Name |
+| [deleteSkillsetByObject.js][deleteSkillsetByObject] | Deletes a Skillset by Object |
+| [getSkillset.js][getSkillset] | Gets a skillset |
+| [listSkillsets.js][listSkillsets] | List all skillsets |
+| [listSkillsetsNames.js][listSkillsetsNames] | List names of skillsets |
+| **SynonymMaps** |
+| [createOrUpdateSynonymMap.js][createOrUpdateSynonymMap] | Updates a synonym map |
+| [createSynonymMap.js][createSynonymMap] | Creates a synonym map |
+| [deleteSynonymMapByName.js][deleteSynonymMapByName] | Deletes a synonym map by Name |
+| [deleteSynonymMapByObject.js][deleteSynonymMapByObject] | Deletes a synonym map by object |
+| [getSynonymMap.js][getSynonymMap] | Gets a synonym map |
+| [listSynonymMapNames.js][listSynonymMapNames] | List names of synonym maps |
+| [listSynonymMaps.js][listSynonymMaps] | List all synonym maps |
```

Comment:
It looks like we are adding a sample for each api exposed in the package. Was this intentional? Are we doing this across languages?

Usually samples would cover either end to end scenarios or the most commonly used apis

Reply:
Am not saying we change anything now, just curious about the reasoning here

Reply:
1. Originally, I added some samples in the readme. The readme review team wanted to keep the samples in the readme to a minimum and said I can move it to samples. 
2. During the user study, I saw 2 people actually searching for sample with keywords in readme. They were looking for sample that does smaller tasks. So, I thought I could add one for each API. (Also I prefer smaller samples which explains a single API vs a larger sample that I need to go through entirely to understand one API that I want :-)) 

I am open to changing it to one large sample which explains all APIs at a single shot too....

Reply:
Collapsing everything to a single sample would not be an ideal situation either.
If we have samples for each api, then the question arises as to when do we expect someone to look at samples vs api reference docs (either on the website or in the editor).

If the idea is to have an example for each api, then the reference docs is a better place.
Keyvault is a good example. See https://github.com/Azure/azure-sdk-for-js/blob/%40azure/keyvault-keys_4.0.4/sdk/keyvault/keyvault-keys/src/index.ts#L297-L312

Samples is also something that we should try and have parity across languages. What are the other languages doing in this area?

Reply:
.NET has a different strategy. They have a readme file wilth samples in between.. https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/search/Azure.Search.Documents/samples ... I kind of like it...

Reply:
> Since this maps to what we saw for TS, I think it's good.
I do agree with Ramya that we should think about where we might consolidate some samples or pluck out very compact examples into reference doc code comments. It may also be useful to align to other languages here. I don't think that work is GA blocking though!

I agree that my comment does not block GA and we should merge this PR so that TS & JS samples are in sync.
My concern is on the follow ups

.Net may not be the right example to follow here due to their sample story being so entirely different from other languages. What are our friends in Java & Python doing?